controllers/cluster.weave.works/existinginframachine_controller.go

```diff
@@ -1265,30 +1282,34 @@ func (m MachineMapper) Map(mo handler.MapObject) []reconcile.Request {
 		}
 	}
 	log.Info("Cluster configuration changed; marking machines as needing repaving")
-	if err := m.reconciler.updateAPIServerArgs(context.TODO(), &eic.Spec.APIServer.ExtraArguments); err != nil {
+	if err := m.reconciler.updateAPIServerArgs(ctx, &eic.Spec.APIServer.ExtraArguments); err != nil {
 		log.Errorf("failed to update API server args: %v", err)
 		return nil
 	}
 
-	// Find the machines needing update
-	result := []reconcile.Request{}
-	machineBytes := []byte(cmap.Data["machines"])
-	var ips []string
-	if err := yaml.Unmarshal(machineBytes, &ips); err != nil {
+	// Find the machines needing update and mark them
+	var machines existinginfrav1.ExistingInfraMachineList
+	if err := m.reconciler.Client.List(ctx, &machines); err != nil {
+		log.Errorf("failed to list machines: %v", err)
 		return nil
 	}
-	for _, ip := range ips {
-		m, err := m.reconciler.findMachineByPrivateAddress(context.TODO(), ip)
+
+	result := []reconcile.Request{}
+	for _, machine := range machines.Items {
+		privateAddress := machine.Spec.Private.Address
+		log.Infof("Marking: %s for repaving", privateAddress)
+		node, err := m.reconciler.findNodeByPrivateAddress(ctx, privateAddress)
 		if err != nil {
+			log.Errorf("Couldn't find node matching machine: %s", machine.Name)
 			continue
 		}
-		result = append(result, reconcile.Request{NamespacedName: types.NamespacedName{Namespace: m.Namespace, Name: m.Name}})
+		m.reconciler.setNodeAnnotation(ctx, node.Name, recipe.PlanKey, `{"cluster": "modified"}`)
+		result = append(result, reconcile.Request{NamespacedName: types.NamespacedName{Namespace: machine.Namespace, Name: machine.Name}})
 	}
 
 	// Update the config map with the new spec and clear the list of machines (they will get added back after repaving)
```

Comment:
We are no longer clearing the machines list in this function.

Reply:
Thank you!